### DISCUSSION TITLE
Arbitrary Slurm options

### DISCUSSION THREAD

**Discussion by multimeric** [wlandau/crew.cluster#12] [Q&A]
Looking at the arguments to [`crew_controller_slurm`](https://wlandau.github.io/crew.cluster/reference/crew_controller_slurm.html), I notice that there is no argument like `sbatch_options` for specifying arbitrary `#SBATCH` options. I suspect this is needed because there are too many `sbatch` flags to ever make an argument for each one (like `slurm_cpus_per_task`). It seems that [luckily `script_lines` are inserted just after the `#SBATCH` lines](https://github.com/wlandau/crew.cluster/blob/7d94ec706c365ea8ac564499fe3bb826e75756ba/R/crew_launcher_slurm.R#L269), so it should be possible to inject more using this parameter. Is this the recommended approach? Or should there be a dedicated parameter for this? I think an `sbatch_options` that takes a named character vector of options would be nice, and may even remove the need for `slurm_cpus_per_task` and co.

**Comment by wlandau** [comment]
Yes, it is recommended to use `script_lines` for injecting SLURM options, e.g. `script_lines = c("#SBATCH ...", "#SBATCH ...")`.
